File: QytetetJava/src/modeloqytetet/Calle.java
////////////////////////////////////////////////////////////////////////////////
// Autor: Alejandro Manzanares Lemus
// 2º GII A1
////////////////////////////////////////////////////////////////////////////////
package modeloqytetet;

public class Calle extends Casilla {
    //Atributos 
    private TituloPropiedad titulo; 

    //Constructor
    public Calle(int numero, TituloPropiedad titulo) {
        super(numero);
        setCoste(titulo.getPrecioCompra());
        this.titulo = titulo;
    }
        
    //Metodos
    public TituloPropiedad asignarPropietario(Jugador jugador){
        titulo.setPropietario(jugador);
        return titulo;
    }
    @Override
    protected TipoCasilla getTipo() {
        return TipoCasilla.CALLE;
    }
    @Override
    protected TituloPropiedad getTitulo() {
        return titulo;
    }
    public int pagarAlquiler(){
        int costeAlquiler = titulo.pagarAlquiler();
        return costeAlquiler;
    }
    private void setTitulo(TituloPropiedad title) {
        titulo = title;
    }
    @Override
    protected boolean soyEdificable(){
        return true;
    }
    public boolean tengoPropietario(){
        return titulo.tengoPropietario();
    }
    @Override
    public String toString( ){
        return super.toString() + "Tipo: " + TipoCasilla.CALLE + "\n"
                + "Titulo: " + titulo;
    }
}

File: QytetetRuby/lib/calle.rb
#///////////////////////////////////////////////////////////////////////////////
#// Autor: Alejandro Manzanares Lemus
#// 2º GII A1
#///////////////////////////////////////////////////////////////////////////////
module ModeloQytetet
class Calle < Casilla

  def initialize(numero, titulo)
    type = TipoCasilla::CALLE
    super(numero, type)
    @coste = titulo.precio_compra
    @titulo = titulo
  end
  
  attr_accessor :titulo
  
  def tengo_propietario()
      return @titulo.tengo_propietario
  end 
  def asignar_propietario(jugador)
      @titulo.propietario = jugador
      return @titulo
  end
  def pagar_alquiler()
    coste_alquiler = @titulo.pagar_alquiler()
  end
  def propietario_encarcelado()
    return @titulo.propietario_encarcelado
  end
  def to_s()
    imprimir = super + "Titulo:#{@titulo}"
    imprimir
  end
  
  private :titulo=
end
end